src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -186,6 +189,16 @@ private async Task ProcessGetBlocksAsync(INetworkPeer peer, GetBlocksPayload get
         {
             this.logger.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(peer), peer.RemoteSocketEndpoint, nameof(getBlocksPayload), getBlocksPayload);
 
+            if (getBlocksPayload.BlockLocators.Blocks.Count > ConsensusManagerBehavior.MaxLocatorSize)
+            {
+                this.logger.LogTrace("Peer '{0}' sent getblocks with oversized locator, disconnecting.", peer.RemoteSocketEndpoint);
+
+                this.peerBanning.BanAndDisconnectPeer(peer.RemoteSocketEndpoint, 0);
```

Comment:
Why zero?

Reply:
If you only want to disconnect, just disconnect the peer, don't introduce peer banning dependency